File: src/main/java/com/leijendary/spring/boot/template/core/config/properties/AuthProperties.java
package com.leijendary.spring.boot.template.core.config.properties;

import org.springframework.boot.context.properties.ConfigurationProperties;

import lombok.Data;

@ConfigurationProperties(prefix = "auth")
@Data
public class AuthProperties {

    private String audience;
    private String realm;
    private AnonymousUser anonymousUser;

    @Data
    public static class AnonymousUser {

        private String principal;
    }
}

File: src/main/kotlin/com/leijendary/spring/boot/template/core/config/properties/AuthProperties.kt
package com.leijendary.spring.boot.template.core.config.properties

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "auth")
class AuthProperties {
    var audience: String = ""
    var realm: String = ""
    var anonymousUser: AnonymousUser = AnonymousUser()
    var system: System = System()

    inner class AnonymousUser {
        var principal: String = ""
    }

    inner class System {
        var principal: String = ""
    }
}